### ISSUE TITLE
Subscribers with topic filtering declared deeper in DOM hierarchy to share an MQTT connection

### ISSUE THREAD

**Issue by gautaz** [telecomsante/tsante-mqtt#3]
The point of this enhancement would be to support this kind of document:

```html
<tsante-mqtt>
  <my-component>
    <tsante-mqtt-subscriber topic="topic1"></tsante-mqtt-subscriber>
  </my-component>
  <my-component>
    <tsante-mqtt-subscriber topic="topic2"></tsante-mqtt-subscriber>
    <tsante-mqtt-subscriber topic="topic3"></tsante-mqtt-subscriber>
  </my-component>
</tsante-mqtt>
```

Each `<tsante-mqtt-subscriber>` would fire their own `tsante-mqtt-received` events by filtering messages matching their `topic` (received by the MQTT client embedded in `<tsante-mqtt>`).

Thus both `<my-component>`s could handle two different set of topics (also taking into account topic wildcards) through the same MQTT connection.

**Comment by fablec-telecomsante** [comment]
due to the use of the Paho client library, the received and delivered events are catched by the connection (tsante-mqtt).

Furthermore, as you must write a javascript method to treat received messages, you can easily distinguish the 2 cases in this function.

**Comment by gautaz** [comment]
Yes, but I won't be able to use the same connection to fire events to be received by two different components as detailed in the previous example.

The only option for now seems to be this:
```html
<my-component>
  <tsante-mqtt>
    <tsante-mqtt-subscriber topic="topic1"></tsante-mqtt-subscriber>
  </tsante-mqtt>
</my-component>
<my-component>
  <tsante-mqtt>
    <tsante-mqtt-subscriber topic="topic2"></tsante-mqtt-subscriber>
    <tsante-mqtt-subscriber topic="topic3"></tsante-mqtt-subscriber>
  </tsante-mqtt>
</my-component>
```

Do you think of any other way to deal with this?

Is there a way to use the same connection and receive MQTT events from it in unrelated parts of the DOM?

**Comment by gautaz** [comment]
@fablec-telecomsante Previous questions remain unanswered and as discussed IRL this morning, the component should still work with subscribers declared deeper in the hierarchy.

Will you accept a PR providing this feature?

**Comment by fablec-telecomsante** [comment]
Yes of course

**Comment by fablec-telecomsante** [comment]
intergrated in v1.4.0

**Comment by gautaz** [comment]
Fixed by #11
